.github/workflows/function-composition.yml

```diff
@@ -361,3 +395,97 @@ jobs:
         if: ${{ always() }}
         run: |
           kn service delete --all --namespace default --wait
+
+  bench-stacking-training-compose:
+    name: Stacking Training (docker-compose)
+    needs: push-stacking-training
+    runs-on: ubuntu-18.04
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        transfer-type: [ s3 ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: 'true'
+
+      - name: start multi-container program with docker-compose
+        env:
+          AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+          AWS_SECRET_KEY: ${{ secrets.AWS_SECRET_KEY }}
+        run: |
+          cd ./function-images/tests/stacking-training/
+          docker-compose -f docker-compose-${{ matrix.transfer-type }}.yml up &> log_file &
+          sleep 1m
+      - name: invoke the chain
+        run: |
+          ./bin/grpcurl -plaintext localhost:50051 helloworld.Greeter.SayHello
+      - name: show docker-compose log
+        if: ${{ always() }}
+        run: cat  ./function-images/tests/stacking-training/log_file
+
+  bench-stacking-training-knative:
+    name: Stacking Training (Knative)
+    needs: push-stacking-training
+    timeout-minutes: 30
+    env:
+      GOCACHE: /root/tmp/gocache
+      GOPATH: /root/tmp/gopath
+      TMPDIR: /root/tmp
+    runs-on: [ stock-knative ]
+    strategy:
+      fail-fast: false
+      #matrix:
+        #transfer-type: [ s3 ]
+    steps:
+      - name: Setup TMPDIR
+        run: mkdir -p $TMPDIR
+
+      - uses: actions/checkout@v2
+        with:
+          lfs: 'true'
+
+      - name: Deploy functions
+        env:
+          AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
+          AWS_SECRET_KEY: ${{ secrets.AWS_SECRET_KEY }}
+        run: |
+          ./function-images/tests/kn_deploy.sh ./function-images/tests/stacking-training/manifests/s3/service*
+          sleep 1m
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Build invoker
+        working-directory: examples/invoker
+        run: make invoker
+
+      - name: Invoke functions
+        working-directory: examples/invoker
+        run: |
+          set -x
+
+          NODEPORT=$(80)
+          HOSTNAME=$(kubectl get ksvc streaming -n default -o jsonpath='{.status.url}' | cut -c8-)
+
+          echo '[ { "hostname": "$HOSTNAME" } ]' > endpoints.json
+          ./invoker -port $NODEPORT -dbg -rps 0.3
```

Comment:
the invoker manages to invoke it 2x, please adjust its runtime so it waits for 2 min (remove the sleep though) and `-rps` so that it only makes a single invocation